bin/actions-toolkit.js

```diff
@@ -1,47 +1,134 @@
+        choices: colors
+      }
+    ])
+  } catch (err) {
+    // When prompt() rejects, that means the user has pressed ctrl+c to cancel input.
+    throw new Error('Cancelled. Maybe next time!')
   }
-  process.exit(1)
 }
 
-const { version } = require('../package.json')
-const packageJson = {
-  name,
-  private: true,
-  main: 'index.js',
-  dependencies: {
-    'actions-toolkit': `^${version}`
+/**
+ *
+ * @param {PromptOptions} options
+ * @returns {Promise<string>} The Dockerfile contents.
+ */
+const createDockerfile = async (options) => {
+  const dockerfileTemplate = await readFile(path.join(templateDir, 'Dockerfile'), 'utf8')
+  return dockerfileTemplate
+    .replace(':NAME', options.name)
+    .replace(':DESCRIPTION', options.description)
+    .replace(':ICON', options.icon)
+    .replace(':COLOR', options.color)
+}
+
+const createPackageJson = (name) => {
+  const { version } = require('../package.json')
+  return {
+    name,
+    private: true,
+    main: 'index.js',
+    dependencies: {
+      'actions-toolkit': `^${version}`
+    }
   }
 }
 
-const templateDir = path.join(__dirname, 'template')
+const runCLI = async (argv) => {
+  const args = minimist(argv)
+  const directoryName = args._[0]
+  if (!directoryName || args.help) {
+    console.log(`\nUsage: npx actions-toolkit <name>`)
+    process.exit(1)
+    return
+  }
+  const base = path.join(process.cwd(), directoryName)
+  try {
+    console.log(`Creating folder ${base}...`)
+    await mkdir(base)
+  } catch (err) {
+    if (err.code === 'EEXIST') {
+      throw new Error(`Folder ${base} already exists!`)
+    } else {
+      console.error(err.code)
+      throw err
+    }
+  }
 
-console.log('Creating package.json...')
-fs.writeFileSync(path.join(base, 'package.json'), JSON.stringify(packageJson, null, 2))
+  console.log("\nWelcome to actions-toolkit! Let's get started creating an action.")
+  const metadata = await getActionMetadata()
+  const dockerfile = await createDockerfile(metadata)
+  const packageJson = createPackageJson(directoryName)
+  const entrypoint = await readFile(path.join(templateDir, 'entrypoint.js'), 'utf8')
+  const files = [
+    ['package.json', JSON.stringify(packageJson, null, 2)],
+    ['Dockerfile', dockerfile],
+    ['entrypoint.js', entrypoint]
+  ]
+  files.forEach(async ([filename, contents]) => {
+    console.log(`Creating ${filename}...`)
+    await writeFile(path.join(base, filename), contents)
+  })
 
-fs.readdirSync(templateDir).forEach(file => {
-  const contents = fs.readFileSync(path.join(templateDir, file))
-  console.log(`Creating ${file}...`)
-  fs.writeFileSync(path.join(base, file), contents)
-})
+  console.log(`\nDone! Enjoy building your GitHub Action! Get started with:\n\ncd ${directoryName} && npm install`)
+}
 
-console.log(`\nDone! Enjoy building your GitHub Action! Get started with:\n\ncd ${name} && npm install`)
+module.exports = runCLI
```

Comment:
So right now, this is a module for easier testability. I'm wondering if I should split this into another file called `run.js` or something like that and import it here? Or maybe make a file called `bin/cli` that requires this function and runs:

```js
const run = require('./run')

run(process.argv.slice(2))
  .then(() => {
    // success
  })
  .catch(err => {
    console.error(err)
    process.exit(1)
  })
```

What do you think of this architecturally?

Reply:
Yeah I like that! Looks great 👌 Thanks for chatting through the decisions!